tokens/usecase/tokens_usecase.go

```diff
@@ -246,8 +253,18 @@ func (t *tokensUseCase) getPricesForBaseDenom(ctx context.Context, baseDenom str
 		wg.Add(1)
 		go func(baseDenom, quoteDenom string) {
 			defer wg.Done()
-
-			price, err := pricingStrategy.GetPrice(ctx, baseDenom, quoteDenom, pricingOptions...)
+			var price osmomath.BigDec
+			var err error
+			price, err = pricingStrategy.GetPrice(ctx, baseDenom, quoteDenom, pricingOptions...)
+			if err != nil { // Check if we should fallback to another pricing source
+				fallbackSourceType := pricingStrategy.ShouldFallback(quoteDenom)
+				if fallbackSourceType != domain.NoneSourceType {
+					fallbackPricingStrategy, ok := t.pricingStrategyMap[fallbackSourceType]
+					if ok {
+						price, err = fallbackPricingStrategy.GetPrice(ctx, baseDenom, quoteDenom, pricingOptions...)
+					}
+				}
```

Comment:
Let's try to cover this with a test case. We can do so by creating a mock of `PricingSource` and setting up the mock as to force fallback.

See example of how these mocks are used:
https://github.com/osmosis-labs/sqs/tree/v24.x/domain/mocks

We can do this similarly